File: 3550-61-1358-number-of-substrings-containing-all-three-characters/3550-61-1358-number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int vis[3]={-1,-1,-1},cnt=0;
        for(int i = 0 ; i < s.size();i++){
            vis[s[i]-'a'] = i; 
            cnt+=min({vis[0],vis[1],vis[2]})+1; 
        }
        return cnt;
    }
};

File: 3550-61-1358-number-of-substrings-containing-all-three-characters/3550-61-1358-number-of-substrings-containing-all-three-characters.java
class Solution {
    public int numberOfSubstrings(String s) {
        int[] vis = {-1, -1, -1}; 
        int cnt = 0; 

        for (int i = 0; i < s.length(); i++) {
            vis[s.charAt(i) - 'a'] = i; 
            int minIndex = Math.min(vis[0], Math.min(vis[1], vis[2]));
            cnt += minIndex + 1;
        }

        return cnt;
    }
}